src/commons/assessmentWorkspace/AssessmentWorkspace.tsx

```diff
@@ -166,6 +205,46 @@ class AssessmentWorkspace extends React.Component<
     this.checkWorkspaceReset(this.props);
   }
 
+  public uploadPerHour = () => {
+    const interval = setInterval(() => {
+      this.uploadLogs();
+    }, oneHourInMilliSeconds);
+    return () => clearInterval(interval);
+  };
+
+  public uploadLogs = () => {
+    const assessmentIDs = getLoggedAssessmentIds();
+    const assessmentLogs = getAssessmentLogs();
+    const playgroundLogs = getPlaygroundLogs();
+
+    if (assessmentLogs.inputs.length !== 0) {
+      this.props.handleKeystrokeUpload(
+        assessmentIDs.assessmentId,
+        assessmentIDs.questionId,
+        assessmentLogs
+      );
+    }
```

Comment:
Have you tried uploading logs when the system is offline? 

Does error propagation prevent logs from being reset?

Reply:
Yep, there is supposed to be some form of mechanism which helps to do so. 

Unfortunately, I think the stuff I wrote didn't get committed >< 

let me commit it now!